tests/compass-runtime-agent/test/runtimeagent/test/runtime_agent_test.go

```diff
@@ -47,315 +51,283 @@ const (
+			description: "Test case 1: Create all types of API packages and remove them",
+			log:         testkit.NewLogger(t, map[string]string{"ApplicationName": "test-app-1"}),
 			initialPhaseInput: func() *applications.ApplicationRegisterInput {
 				return applications.NewApplication("test-app-1", "provider 1", "testApp1", map[string]interface{}{}).
-					WithAPIDefinitions(
-						[]*applications.APIDefinitionInput{
-							noAuthAPIInput,
-							basicAuthAPIInput,
-							oauthAPIInput,
-						}).
-					WithEventDefinitions(
-						[]*applications.EventDefinitionInput{
-							applications.NewEventDefinition("events-api", "description").WithJsonEventSpec(&apiSpecData),
-							applications.NewEventDefinition("events-api-with-empty-string-spec", "description").WithYamlEventSpec(&emptySpec),
-							applications.NewEventDefinition("no-description-events-api", "").WithYamlEventSpec(&apiSpecData),
-						},
-					)
+					WithAPIPackages(noAuthAPIPkgInput, basicAuthAPIPkgInput, oauthAPIPkgInput, basicAuthWithCSRFPkgInput)
 			},
 			initialPhaseAssert: assertK8sResourcesAndAPIAccess,
-			secondPhaseSetup: func(t *testing.T, testSuite *runtimeagent.TestSuite, this *testCase) {
+			secondPhaseSetup: func(t *testing.T, testSuite *runtimeagent.TestSuite, this *TestCase) {
 				// when removing all APIs individually
 				application := this.initialPhaseResult.Application
-				assert.Equal(t, 3, len(application.APIDefinitions.Data))
-				assert.Equal(t, 3, len(application.EventDefinitions.Data))
-
-				// remove APIs
-				for _, api := range application.APIDefinitions.Data {
-					id, err := testSuite.CompassClient.DeleteAPI(api.ID)
-					require.NoError(t, err)
-					require.Equal(t, api.ID, id)
-				}
+				apiPackages := application.Packages.Data
+				require.Equal(t, 4, len(apiPackages))
 
-				// remove EventAPIs
-				for _, eventAPI := range application.EventDefinitions.Data {
-					id, err := testSuite.CompassClient.DeleteEventAPI(eventAPI.ID)
+				// remove Packages
+				for _, pkg := range application.Packages.Data {
+					id, err := testSuite.CompassClient.DeleteAPIPackage(pkg.ID)
 					require.NoError(t, err)
-					require.Equal(t, eventAPI.ID, id)
+					require.Equal(t, pkg.ID, id)
 				}
 
 				// then
-				this.secondPhaseAssert = func(t *testing.T, testSuite *runtimeagent.TestSuite, this *testCase) {
+				this.secondPhaseAssert = func(t *testing.T, testSuite *runtimeagent.TestSuite, this *TestCase) {
 					// assert APIs deleted
-					for _, api := range application.APIDefinitions.Data {
-						testSuite.K8sResourceChecker.AssertAPIResourcesDeleted(t, application.Name, api.ID)
-					}
-
-					// assert EventAPIs deleted
-					for _, eventAPI := range application.EventDefinitions.Data {
-						testSuite.K8sResourceChecker.AssertAPIResourcesDeleted(t, application.Name, eventAPI.ID)
+					for _, pkg := range apiPackages {
+						testSuite.K8sResourceChecker.AssertAPIPackageDeleted(t, this.log, pkg, application.Name)
```

Comment:
It seems that AssertAPIPackageDeleted checks only Application CRD contents and doesn't verify whether ClusterAssetGroup was deleted.